vm/src/protocol/iter.rs

```diff
@@ -14,6 +14,13 @@ pub struct PyIter<O = PyObjectRef>(O)
 where
     O: Borrow<PyObject>;
 
+#[cfg(feature = "gc")]
+unsafe impl<O: crate::object::Trace + Borrow<PyObject>> crate::object::Trace for PyIter<O> {
+    fn trace(&self, tracer_fn: &mut crate::object::TracerFn) {
+        self.0.trace(tracer_fn);
```

Comment:
Shouldn't this just `self.0.borrow().trace()`? Shouldn't need the Trace bound I think.